pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -4633,3 +4634,103 @@ func TestPendingDeleteReplacement(t *testing.T) {
+				DiffF: func(urn resource.URN, id resource.ID,
+					olds, news resource.PropertyMap, ignoreChanges []string,
+				) (plugin.DiffResult, error) {
+					return plugin.DiffResult{Changes: plugin.DiffSome}, nil
+				},
+				UpdateF: func(_ resource.URN, _ resource.ID, _, _ resource.PropertyMap, _ float64,
+					_ []string, _ bool,
+				) (resource.PropertyMap, resource.Status, error) {
+					outputs := resource.NewPropertyMapFromMap(map[string]interface{}{
+						"foo": "bar",
+					})
+					return outputs, resource.StatusOK, nil
+				},
+			}, nil
+		}, deploytest.WithoutGrpc),
+	}
+
+	program := deploytest.NewLanguageRuntime(func(info plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		_, _, _, err := monitor.RegisterResource(
+			resource.RootStackType,
+			info.Project+"-"+info.Stack,
+			false,
+			deploytest.ResourceOptions{})
+		require.NoError(t, err)
+
+		_, _, _, err = monitor.RegisterResource(
+			"pkgA:m:typA",
+			"resA",
+			true,
+			deploytest.ResourceOptions{})
+		require.NoError(t, err)
+
+		return nil
+	})
+
+	p.Options.Host = deploytest.NewPluginHost(nil, nil, program, loaders...)
+
+	// Run an update to create the resource -- created and updated should be set and equal.
+	p.Steps = []TestStep{{Op: Update, SkipPreview: true}}
+	snap := p.Run(t, nil)
+	creationTimes := []*time.Time{}
+	require.NotEmpty(t, snap.Resources)
+	for _, resource := range snap.Resources {
+		creationTimes = append(creationTimes, resource.Created)
+
+		tz, _ := resource.Created.Zone()
+		assert.Equal(t, "UTC", tz)
+		assert.NotNil(t, resource.Created, resource.URN)
+		assert.NotNil(t, resource.Updated, resource.URN)
+		assert.Equal(t, resource.Created, resource.Updated, resource.URN)
+	}
+
+	// Run a refresh -- created and updated should be unchanged.
+	p.Steps = []TestStep{{Op: Refresh, SkipPreview: true}}
+	snap = p.Run(t, snap)
+	require.NotEmpty(t, snap.Resources)
+	for _, resource := range snap.Resources {
+		assert.NotNil(t, resource.Created, resource.URN)
+		assert.NotNil(t, resource.Updated, resource.URN)
+		assert.Equal(t, resource.Created, resource.Updated, resource.URN)
+		assert.Contains(t, creationTimes, resource.Created, resource.URN)
+	}
+
+	// Run another update -- updated should be greater than created for resA,
+	// everything else should be untouched.
+	p.Steps = []TestStep{{Op: Update, SkipPreview: true}}
+	snap = p.Run(t, snap)
+	require.NotEmpty(t, snap.Resources)
+	for _, resource := range snap.Resources {
+		assert.NotNil(t, resource.Created, resource.URN)
+		assert.NotNil(t, resource.Updated, resource.URN)
+		assert.Contains(t, creationTimes, resource.Created, resource.URN)
+
+		switch resource.Type {
+		case "pkgA:m:typA":
+			tz, _ := resource.Updated.Zone()
+			assert.Equal(t, "UTC", tz)
+			assert.Greater(t, *resource.Updated, *resource.Created, resource.URN)
+			assert.NotContains(t, creationTimes, resource.Updated, resource.URN)
+		case "pulumi:providers:pkgA", "pulumi:pulumi:Stack":
+			// Nothing else to do.
```

Comment:
These should probably assert that they are still equal. We've been bitten a few times by not testing the default case and only testing things we expected to change.